spec/inputs/http_poller_spec.rb

```diff
@@ -2,87 +2,151 @@
 require 'logstash/inputs/http_poller'
 
 describe LogStash::Inputs::HTTP_Poller do
-  let(:settings) {
+  let(:queue) { Queue.new }
+  let(:default_interval) { 5 }
+  let(:default_name) { "url1 " }
+  let(:default_url) { "http://localhost:1827" }
+  let(:default_urls) {
     {
-      "interval" => 10,
-      "request_timeout" => 1,
-      "connect_timeout" => 1,
-      "socket_timeout" => 1,
-      "automatic_retries" =>1,
-      "type" => "something-fancy"
+      default_name => default_url
     }
   }
-  context "with json codec, when remote server is failing" do
-    it "should not add an event" do
-      subject = LogStash::Inputs::HTTP_Poller.new(settings.merge({"urls" => {"url1" => "http://mytesturl.mydomain"}}))
-      # no stub, and an incorrect domain, this should result in no response.
+  let(:default_opts) {
+    {
+      "interval" => default_interval,
+      "urls" => default_urls,
+      "codec" => "json"
+    }
+  }
+  let(:mod) { LogStash::Inputs::HTTP_Poller }
 
-      logstash_queue = Queue.new
-      subject.run_once logstash_queue
+  subject { mod.new(default_opts) }
+
+  describe "#run" do
+    it "should run at the specified interval" do
+      expect(Stud).to receive(:interval).with(default_interval).once
+      subject.send(:run, double("queue"))
```

Comment:
ooh, good call. Let's add a comment there to note that you're using send to call the private method (lol private, ruby!)